airflow/providers/postgres/hooks/postgres.py

```diff
@@ -184,7 +184,8 @@ def get_iam_token(self, conn: Connection) -> Tuple[str, str, int]:
             # Pull the custer-identifier from the beginning of the Redshift URL
             # ex. my-cluster.ccdre4hpd39h.us-east-1.redshift.amazonaws.com returns my-cluster
             cluster_identifier = conn.extra_dejson.get('cluster-identifier', conn.host.split('.')[0])
-            client = aws_hook.get_client_type('redshift')
+            session, endpoint_url = aws_hook._get_credentials()
+            client = session.client('redshift', endpoint_url=endpoint_url, config=aws_hook.config, verify=aws_hook.verify)
```

Comment:
Is this the right fix? I thought the issue is only when using `get_client_type('iam')` ?
Shouldn't we set the `client_type` on the hook and then use `aws_hook.get_client_type(client_type='redshift')` ?

Reply:
get_client_type barks with Deprecation on any first argument ever passed. This and glue are among other references that I found in IDE just now. I said I haven't investigated entire codebase for similar gaps earlier...

Reply:
Reread the code fragment - The above mentioned code piece is utilizing hook for client_type 'rds', so I think i probably shouldn't be changing the client_type it has in hooke constructor, as it will break the non-Redshift scenario in `else` condition (potentially).